torch/_functorch/aot_autograd.py

```diff
@@ -1700,7 +1713,11 @@ def functionalized_f_helper(*args):
                     # Since keep_input_mutations is set, we need to faithfully apply a copy_()
                     # so the compiler will see the input mutation in the graph.
                     assert inpt_new is not inpt_old
-                    inpt_old.copy_(inpt_new)
+                    if meta.input_info[i].mutations_triton_only:
+                        #with torch.no_grad():
```

Comment:
Ah locally when I switch to this and look at the error, it looks like it's not coming from this line - it's coming from inductor's FakeTensorProp: https://github.com/pytorch/pytorch/blob/18cc8a92aca8529b9c689b1c64f70bf86509e258/torch/_inductor/compile_fx.py#L525.

This is because inductor's FakeTensorProp is re-running the graph, including the `copy_()`, and it's not running it under no_grad so we get the same error.

Reply:
This should be enough to fix it:
```
diff --git a/torch/_inductor/compile_fx.py b/torch/_inductor/compile_fx.py
index 90d3ef7afae..223ceb84489 100644
--- a/torch/_inductor/compile_fx.py
+++ b/torch/_inductor/compile_fx.py
@@ -521,7 +521,8 @@ def fx_codegen_and_compile(
     # .view() call.
     view_to_reshape(gm)

-    fake_mode = fake_tensor_prop(gm, example_inputs)
+    with torch.no_grad():
+        fake_mode = fake_tensor_prop(gm, example_inputs)

     # pattern matcher passes might not preserve striding information
```

One reasonable question: is it valid to run the FakeTensorProp code under no_grad? Well, by the time we're in inductor, we assume that AOTAutograd has already "taken care" of autograd, so there should be no more autograd-related API's in the graph. Running under no_grad() could in theory supress valid errors that autograd might want to raise, but we should be guaranteed that if this code were invalid, AOTAutograd would have already raised an error on it.